### PR TITLE
Add rate limit to pressing 'enter' in IC to prevent accidental blankposting

### LINKED ISSUE #287: It's far too easy to accidentally send a blank post when sending short messages.
To reproduce: Hold enter for a short period after sending a short message. It will be followed up by a blank post.
Solution: Either limit the rate of sent messages on the tsuserver end or on the client end.

I accidentally do this all the time post-update and it's infuriating.

### PR DESCRIPTION
**Pull request by in1tiate**
Fixes #287

### CODE REVIEW

File: src/courtroom.cpp [hunk 1]
```diff
@@ -1525,6 +1525,8 @@ void Courtroom::on_chat_return_pressed()
   if ((anim_state < 3 || text_state < 2) && objection_state == 0)
     return;
 
+  ui_ic_chat_message->blockSignals(true);
+  QTimer::singleShot(600, this, SLOT(ratelimit_ic()));
```
Comment (oldmud0) [inline src/courtroom.cpp:1529]: You can do this without an additional function:

```suggestion
  QTimer::singleShot(600, this, [=] {
    ui_ic_chat_message->blockSignals(false);
  });
```

Comment (in1tiate) [reply src/courtroom.cpp:1529]: Excellent, thank you.

**Comment by oldmud0** [comment]
Tested?

**Comment by in1tiate** [comment]
Tested briefly on Official Unofficial, this works as of 6c2010a

**Review by oldmud0 — APPROVED** [review]

**Review by Crystalwarrior — APPROVED** [review]
Huh, didn't realize you can handle timers like this. Noted...

**Pull request merged**